.github/workflows/copilot-environment.yml

```diff
@@ -0,0 +1,124 @@
+name: Copilot Environment Setup
+
+# This workflow sets up the development environment for GitHub Copilot coding agent
+# It ensures all necessary tools and dependencies are available for automated coding tasks
+#
+# The workflow:
+# - Sets up Rust toolchain as specified in rust-toolchain.toml
+# - Installs and caches Cargo dependencies
+# - Installs additional development tools (cargo-audit, cargo-outdated)
+# - Verifies Azure SDK specific system dependencies (OpenSSL, build tools)
+# - Runs comprehensive workspace verification (check, clippy, fmt)
+# - Tests basic functionality to ensure the environment is ready
+#
+# Triggers:
+# - Manual execution via workflow_dispatch 
+# - Weekly scheduled runs to keep environment fresh
+# - Automatic runs when key configuration files change
+
+on:
+  workflow_dispatch: # Allow manual triggering
+  schedule:
+    # Run weekly to keep the environment fresh
+    - cron: '0 2 * * 1' # Monday at 2 AM UTC
+  push:
+    branches: [main]
+    paths:
+      - 'rust-toolchain.toml'
+      - 'Cargo.toml'
+      - '.github/workflows/copilot-environment.yml'
+
+jobs:
+  setup-environment:
+    name: Setup Copilot Coding Agent Environment
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      - name: Setup Rust toolchain from rust-toolchain.toml
+        uses: dtolnay/rust-toolchain@stable
+        # This action automatically reads rust-toolchain.toml and installs the specified components
+      
+      - name: Verify Rust installation
```

Comment:
You already print all this at the end. Don't waste time doing it twice. Just do it at the end.